packages/lib/src/core/core.ts

```diff
@@ -143,6 +149,18 @@ class Core {
         }
 
         if (action.type) {
+            // Call analytics endpoint
+            const aObj: AnalyticsObject = createAnalyticsObject({
+                class: 'log',
+                component: `${action.type}${action.subtype}`,
+                type: ANALYTICS_ACTION_STR,
+                subtype: capitalizeFirstLetter(action.type),
+                message: `${action.type}${action.subtype} is initiating`
```

Comment:
Can't we somehow move this logic to the Analytics module? This way we keep analytics logic all there and do not spread in the Core/UIElement

```js
analytics.createAnalyticsEvent({ type: 'log', data: { class, component, type, etc.. } });
```

In the analytics module:
```js
createAnalyticsEvent({ type, data }) {
  const analyticsData =  createAnalyticsObject(...data, etc);
  this.addAnalyticsAction(type, analyticsData);
}

Reply:
yeah, good call